js/main.js

```diff
@@ -0,0 +1,125 @@
+/* global data */
+/* exported data */
+
+var $mainHeading = document.querySelector('#main-heading');
+var $gameForm = document.querySelector('form');
+var $categoryWrapper = document.querySelector('#category-wrapper');
+var $difficultyWrapper = document.querySelector('#difficulty-wrapper');
+var categorySelection = '';
+var difficultySelection = '';
+var sessionCode = '';
+
+// HANDLE CATEGORY CLICKS
+function handleCategoryClicks(event) {
+  if (event.target.tagName !== 'INPUT') {
+    return;
+  }
+  var $closestCategory = event.target.closest('[data-category-id]');
+  var categoryID = $closestCategory.getAttribute('data-category-id');
+  categorySelection = categoryID.toString();
+  $categoryWrapper.setAttribute('class', 'row hidden');
+  renderDifficuly();
+}
+
+// CATEGORY CLICK LISTENER
+$categoryWrapper.addEventListener('click', handleCategoryClicks);
+
+// HANDLE DIFFICULTY CLICKS
+function handleDifficultyClicks(event) {
+  if (event.target.tagName !== 'INPUT') {
+    return;
+  }
+  if (event.target.name === 'easy') {
+    difficultySelection = 'easy';
+  } else if (event.target.name === 'med') {
+    difficultySelection = 'medium';
+  } else if (event.target.name === 'hard') {
+    difficultySelection = 'hard';
+  }
+}
+
+$difficultyWrapper.addEventListener('click', handleDifficultyClicks);
+
+// HANDLE FORM
+function handleGameForm(event) {
+  event.preventDefault();
+  // GET SESSION TOKEN
+  var xhrToken = new XMLHttpRequest();
+  xhrToken.open('GET', 'https://opentdb.com/api_token.php?command=request');
+  xhrToken.responseType = 'json';
+  xhrToken.addEventListener('load', function getSessionToken() {
+    console.log('token status', xhrToken.status);
+    console.log('token response', xhrToken.response);
+    var xhrTokenCode = xhrToken.response.token;
+    sessionCode = xhrTokenCode;
+  });
+  xhrToken.send();
+  // GET QUIZ QUESTIONS
+  var xhrGame = new XMLHttpRequest();
+  xhrGame.open('GET', 'https://opentdb.com/api.php?amount=5' + '&' + 'category=' +
+          categorySelection + '&' + 'difficulty=' + difficultySelection + '&' +
+          'type=multiple' + '&' + sessionCode);
+  xhrGame.responseType = 'json';
+  xhrGame.addEventListener('load', function () {
+    console.log('xhr status:', xhrGame.status);
+    console.log('xhr response:', xhrGame.response);
+  });
+  xhrGame.send();
```

Comment:
Does this actually get the session code? You should have this in the `load` event handler for the prior request, the data from that request's response will not have returned and been dealt with before these lines run.